test/TestDataGenerator.cpp

```diff
@@ -0,0 +1,233 @@
+#include "../src/DataGenerator.h"
+#include "../src/utils.h"
+#include "gtest/gtest.h"
+#include <iostream>
+#include <sstream>
+#include <utility>
+
+// TEST(TestDataGenerator, TestDatasetGenerator) {
+//   DataGenerator data = DataGenerator();
+//   data.DatasetGenerator(3, 20, 4, "test.xml");
+// }
+
+// TEST(TestDataGenerator, TestModelGenerator) {
+//   DataGenerator data = DataGenerator();
+//   data.modelGenerator(2, 3, "test_model.xml");
+// }
+
+TEST(TestDataGenerator, TestXmlFileType) {
+  DataGenerator data = DataGenerator();
+  std::string type = data.getXmlFileType("test_model.xml");
```

Comment:
I suppose your test files are saved locally, it's not a problem to include them in the repo as mock data ! Otherwise the pipeline will fail anyway

Reply:
Remove from gitignore data directory.